.scripts/transform_files.py

```diff
@@ -0,0 +1,437 @@
+    )
+    if not release_match:
+        sys.stderr.write(
+            f"release version {release_version} has to use a format like v[0-9]+.[0-9]+.[0-9]+, e.g. v2.6.0)"
+        )
+        sys.exit(2)
+
+    global next_version
+    next_version = read_environment_variable("next_version", False)
+    if next_version == "":
+        v1 = int(release_match["v1"])
+        v2 = int(release_match["v2"])
+        v3 = int(release_match["v3"])
+        next_version = calculate_next_version(release_version, v1, v2, v3)
+
+    global deployed_version
+    deployed_version = read_environment_variable("deployed_version", False)
+    if deployed_version == "":
+        deployed_version = read_next_plantuml_version()
+
+
+def replace_first_regex_in_file(file_path, search, replace):
+    r = re.compile(search)
+    with open(file_path, "r") as file:
+        filedata = file.read()
+        filedata = r.sub(replace, filedata, 1)
+    with open(file_path, "w") as file:
+        file.write(filedata)
+
+
+def replace_in_file(file_path, orig, replace):
+    with open(file_path, "r") as file:
+        filedata = file.read()
+        filedata = filedata.replace(orig, replace)
+    with open(file_path, "w") as file:
+        file.write(filedata)
+
+
+def replace_first_regex_copy_file(
+    source_path, target_path, compiled_search_regex, replace
+):
+    with open(source_path, "r") as file:
+        filedata = file.read()
+    filedata = compiled_search_regex.sub(replace, filedata, 1)
+    with open(target_path, "w") as file:
+        file.write(filedata)
+
+
+# Calculates the next version (inclusive starting v) based on the give version values.
+# If v3 == 0 then v2 is increased otherwise v3
+def calculate_next_version(release, v1, v2, v3):
+    print(f"calculates the next_version based on given release_version {release} ...")
+    if v3 == 0:
+        v2 = v2 + 1
+    else:
+        v3 = v3 + 1
+    version = f"v{v1}.{v2}.{v3}"
+    print(
+        f"The calculated next_version = {version}. It can be used as next_version environment variable with following statement"
+    )
+    print(f"    export next_version={version}")
+    return version
+
+
+# Calculates the next released PlantUML version that it can be used as deployed_version environment variable
+# based on http://www.plantuml.com/plantuml/svg/SoWkIImgAStDuSf8JKn9BL9GBKijAixCpzFGv798pKi1oW00 response
+# This function returns "V" + the parsed version number  (e.g. "V1.2022.16")
+def read_next_plantuml_version():
+    # the idea is that the PlantUML version is extracted out of the svg result of "header %version()".
+    # %version() stores beta of next version.
+    # the returned SVG response stores the version inclusive beta in a text element; e.fg. "...<text ...>1.2022.16beta2</text>..."
+    # and this function returns "V" + the parsed version number  (e.g. "V1.2022.16")
+    print(
+        f"calculates the next deployed_version based on the running PlantUML web server response ..."
```

Comment:
Is there a reason for using `f`? (as no formatting is taking place)

Reply:
no there is non, I fixed it